website/www/site/content/en/blog/deploy-python-pipeline-on-flink-runner.md

```diff
@@ -0,0 +1,857 @@
+
+{{< toc >}}
+
+<!--more-->
+
+## Resources to run a Python Beam pipeline on Flink
+
+We develop an Apache Beam pipeline using the Python SDK and deploy it on an Apache Flink cluster using the Apache Flink runner. While the Flink cluster is created by the Flink Kubernetes Operator, we need two components to run the pipeline on the *Flink runner*: the **job service** and the [**SDK harness**](https://beam.apache.org/documentation/runtime/sdk-harness-config/). Roughly speaking, the job service converts details about a Python pipeline into a format that the Flink runner can understand. The SDK harness executes the Python user code. The Python SDK provides convenience wrappers to manage those components, and it can be utilized by specifying *FlinkRunner* in the pipeline option, for example, `--runner=FlinkRunner`. The *job service* is managed automatically. We rely on our own *SDK harness* as a sidecar container for simplicity. Also, we need the **Java IO Expansion Service**, because the pipeline uses Apache Kafka topics for its data source and sink, and the *Kafka Connector I/O* is developed in Java. Simply put, the expansion service is used to serialize data for the Java SDK.
+
+## Set up the Kafka cluster
+
+An Apache Kafka cluster is deployed using the [Strimzi Operator](https://strimzi.io/) on a [minikube](https://minikube.sigs.k8s.io/docs/) cluster. We install Strimzi version 0.39.0 and Kubernetes version 1.25.3. After the [minikube CLI](https://minikube.sigs.k8s.io/docs/start/) and [Docker](https://www.docker.com/) are installed, you can create a minikube cluster by specifying the desired Kubernetes version. You can find the source code for this blog post can be found in the [**GitHub repository**](https://github.com/jaehyeon-kim/beam-demos/tree/master/beam-deploy).
+
+{{< highlight bash >}}
+minikube start --cpus='max' --memory=20480 \
+  --addons=metrics-server --kubernetes-version=v1.25.3
+{{< /highlight >}}
+
+### Deploy the Strimzi operator
+
+The [**GitHub repository**](https://github.com/jaehyeon-kim/beam-demos/tree/master/beam-deploy) keeps manifest files that you can use to deploy the *Strimzi operator*, Kafka cluster, and Kafka management application. To download a different version of the operator, you can download the relevant manifest file by specifying the desired version. By default, the manifest file assumes that the resources are deployed in the *myproject* namespace. However, because we deploy them in the *default* namespace, we need to change the resource namespace. We change the resource namespace using [sed](https://www.gnu.org/software/sed/manual/sed.html).
```

Comment:
```suggestion
The GitHub repository keeps manifest files that you can use to deploy the Strimzi operator, Kafka cluster, and Kafka management application. To download a different version of the operator, download the relevant manifest file by specifying the version. By default, the manifest file assumes that the resources are deployed in the *myproject* namespace. However, because we deploy them in the *default* namespace, we need to change the resource namespace. We change the resource namespace using [sed](https://www.gnu.org/software/sed/manual/sed.html).
```